lib/mllp/peer.ex

```diff
@@ -0,0 +1,80 @@
+defmodule MLLP.Peer do
+  require Logger
+
+  @type t :: %{
+          :transport => :ranch_tcp | :ranch_ssl,
+          :socket => :ranch_transport.socket(),
+          :client_info => {:inet.ip_address(), :inet.port_number()}
+        }
+
+  @type error_type ::
+          :client_ip_not_allowed
+          | :fail_to_verify_client_cert
+
+  @spec validate(t(), Keyword.t()) :: {:ok, :success} | {:error, error_type()}
+
+  def validate(peer, options) do
```

Comment:
If we went with thee suggestion here : https://github.com/HCA-Healthcare/elixir-mllp/pull/34/files#r769075119 
then we could have one validate/2 function that merely matches on the options.This would allow us to avoid passing true/false around  (boolean blindness),  wdyt?